text/000-rustdoc-cfgs-handling.md

```diff
@@ -0,0 +1,283 @@
+Rustdoc: stabilization of the `doc(cfg*)` attributes
+
+- Features Name: `doc_cfg`
+- Start Date: 2022-12-07
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#43781](https://github.com/rust-lang/rust/issues/43781)
+
+
+# Summary
+[summary]: #summary
+
+This RFC aims at providing rustdoc users the possibility to add visual markers to the rendered documentation to know under which conditions an item is available (currently possible through the following unstable features: `doc_cfg`, `doc_auto_cfg` and `doc_cfg_hide`).
+
+It does not aim to allow having a same item with different `cfg`s to appear more than once in the generated documentation.
+
+It does not aim to document items which are *inactive* under the current configuration (i.e., “`cfg`'ed out”).
+
+# Motivation
+[motivation]: #motivation
+
+The goal of this RFC is to stabilize the possibility to add visual markers to the rendered documentation to know under which conditions an item is available.
+
+Providing this information to users will solve a common issue: “Why can I see this item in the documentation and yet can't use it in my code?”.
+The end goal being to provide this information automatically so that the documentation maintenance cost won't increase.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC proposes to add the following attributes:
+
+  * `#![doc(auto_cfg(enable))]`/`#[doc(auto_cfg(disable))]`
```

Comment:
This is totally a bikeshed comment, so please feel free to ignore it, but I think it would be better if the syntax for this was:

* `#![doc(auto_cfg = true)]`
* `#![doc(auto_cfg = false)]`

This introduces fewer novel names, hints that there is a single boolean state which is being “shadowed”, and is more like the style of optional-feature-enabling used in `Cargo.toml`, whereas `foo(enable)`/`foo(disable)` is not like any precedent I can think of in Rust syntax.

Reply:
Very good point!